File: createHecate.py
#!/usr/bin/env python3
import os
import subprocess
import sys

# Define the utilities and their corresponding script files
UTILITIES = {
    "1": ("Create Backup", "utilities/createBackup.py"),
    "2": ("Create Config Variables", "utilities/creteConfigVariables.py"),
    "3": ("Create EOS Apps", "utilities/createEosApps.py"),
    "4": ("Create http.conf", "utilities/createHttpConf.py"),
    "5": ("Create Docker Compose", "utilities/createDockerCompose.py"),
    "6": ("Create Certificates", "utilities/createCerts.py"),
    "7": ("Restore Config", "utilities/createRestore.py"),
    "q": ("Quit", None)
}

def print_menu():
    print("\n--- Deploy Hecate Utility Wrapper ---\n")
    for key, (desc, _) in UTILITIES.items():
        print(f"{key}. {desc}")
    print()

def run_script(script_path):
    # Check if the script exists
    if not os.path.exists(script_path):
        print(f"Error: {script_path} not found.")
        return

    # Execute the script with the current python interpreter.
    # Alternatively, if these scripts are executable, you can use subprocess.call([script_path])
    try:
        subprocess.run([sys.executable, script_path], check=True)
    except subprocess.CalledProcessError as e:
        print(f"Error: Script {script_path} exited with error code {e.returncode}")
    except Exception as e:
        print(f"An error occurred while running {script_path}: {e}")

def main():
    while True:
        print_menu()
        choice = input("Enter the number of the utility to run (or 'q' to quit): ").strip()

        if choice.lower() == 'q':
            print("Exiting deployHecate.py. Goodbye!")
            break

        if choice in UTILITIES:
            desc, script_path = UTILITIES[choice]
            print(f"\nRunning '{desc}' from {script_path}...\n")
            run_script(script_path)
        else:
            print("Invalid selection. Please try again.")

if __name__ == '__main__':
    main()

File: createHecate.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"os/exec"
	"strings"
)

// Utility represents a menu item
type Utility struct {
	key    string
	desc   string
	script string
}

func printMenu(utilities []Utility) {
	fmt.Println("\n--- Deploy Hecate Utility Wrapper ---\n")
	for _, util := range utilities {
		fmt.Printf("%s. %s\n", util.key, util.desc)
	}
	fmt.Println()
}

func runScript(scriptPath string) {
	// Check if the script exists
	if _, err := os.Stat(scriptPath); os.IsNotExist(err) {
		fmt.Printf("Error: %s not found.\n", scriptPath)
		return
	}

	// Execute the script using Python3
	cmd := exec.Command("python3", scriptPath)
	cmd.Stdout = os.Stdout
	cmd.Stderr = os.Stderr
	if err := cmd.Run(); err != nil {
		// If the error is from the command exiting with a non-zero status
		if exitError, ok := err.(*exec.ExitError); ok {
			fmt.Printf("Error: Script %s exited with error code %d\n", scriptPath, exitError.ExitCode())
		} else {
			fmt.Printf("An error occurred while running %s: %v\n", scriptPath, err)
		}
	}
}

func main() {
	utilities := []Utility{
		{"1", "Create Backup", "utilities/createBackup.go"},
		{"2", "Create Config Variables", "utilities/creteConfigVariables.go"},
		{"3", "Create EOS Apps", "utilities/createEosApps.go"},
		{"4", "Create http.conf", "utilities/createHttpConf.go"},
		{"5", "Create Docker Compose", "utilities/createDockerCompose.go"},
		{"6", "Create Certificates", "utilities/createCerts.go"},
		{"7", "Restore Config", "utilities/createRestore.go"},
		{"q", "Quit", ""},
	}

	reader := bufio.NewReader(os.Stdin)
	for {
		printMenu(utilities)
		fmt.Print("Enter the number of the utility to run (or 'q' to quit): ")
		input, err := reader.ReadString('\n')
		if err != nil {
			fmt.Println("Error reading input, please try again.")
			continue
		}

		choice := strings.TrimSpace(input)
		if strings.ToLower(choice) == "q" {
			fmt.Println("Exiting deployHecate. Goodbye!")
			break
		}

		var found bool
		for _, util := range utilities {
			if util.key == choice {
				found = true
				fmt.Printf("\nRunning '%s' from %s...\n\n", util.desc, util.script)
				runScript(util.script)
				break
			}
		}

		if !found {
			fmt.Println("Invalid selection. Please try again.")
		}
	}
}
